hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala

```diff
@@ -288,6 +288,20 @@ object LowerMatrixIR {
               irRange(0, 'global(lengths)(colIdx), 1).map(Symbol(genUID()) ~> 'row(entriesField)(colIdx)))))
         .mapGlobals('global.dropFields(lengths))
 
+    case MatrixAggregateRowsByKey(child, entryExpr, rowExpr) =>
+      lower(child)
+        .aggregateByKey(
+          aggLet(va = 'row) {
+            rowExpr.insertFields(entriesField -> irRange(0, 'global (colsField).len)
+              .aggElements('__element_idx, '__result_idx)(
+                  let(sa = 'global(colsField)('__result_idx)) {
+                    aggLet(sa = 'global (colsField)('__element_idx),
```

Comment:
Why is `sa` defined twice?

Reply:
We need it once in the aggregation scope, and once in the post-aggregation scope